ttkwidgets/shell.py

```diff
@@ -0,0 +1,205 @@
+        :param prefix: A prefix to show on every input line
+        :type prefix: str
+        :param force_focus: whether or not the shell should take the focus.
+        :type force_focus: bool
+        :param font: Font to use for the terminal
+        :type font: tkinter.font.Font, tuple, or None
+        """
+        for key, value in {
+            'background': 'black',
+            'takefocus': True,
+        }.items():
+            if key not in kwargs:
+                kwargs[key] = value
+        super().__init__(master, **kwargs)
+        if kwargs['takefocus'] and force_focus:
+            self.focus_force()
+
+        self.bind('<Key>', self.on_key_press)
+        self.bind('<Configure>', self.on_configure)
+        self.textvariable = textvariable if isinstance(textvariable, tk.StringVar) else tk.StringVar()
+        self.prefix = prefix
+        self.font = font or ('Terminal', 10)
+        self.line_pos = (5, 5)
+        self.texts = []
+        self.last_text = None
+        self.cursor = None
+        self.blink = True
+        self.buffer = prefix
+        self.text_update()
+        self.commands = defaultdict(list)
+        self.cursor_blink()
+
+    def on_key_press(self, event):
+        if event.keysym == 'Return' and len(self.buffer) > len(self.prefix):
+            self.texts.append(self.last_text)
+            span = self.text_line_span()
+            self.last_text = None
+            self.line_pos = (5, self.line_pos[1] + 15 * span)
+            self.call_command('onreturn', self.buffer[len(self.prefix):])
+            self.buffer = self.prefix
+            self.text_update()
+            return
+
+        if event.keysym == 'Tab':
+            self.call_command('ontab', self)
+            return
+
+        if event.keysym == 'BackSpace' and len(self.buffer) > len(self.prefix):
+            self.buffer = self.buffer[:-1]
+            self.text_update()
+            return
+
+        if event.char.strip() or event.keysym == 'space':
+            self.buffer += event.char
+            self.text_update()
+            return
+
+    def on_configure(self, event):
+        padding = 4
+        width = self.master.winfo_width() - padding
+        height = self.master.winfo_height() - padding
+        self.config(width=width, height=height)
+        for t in self.texts:
+            self.itemconfig(t, width=width)
+        self.itemconfig(self.last_text, width=width)
+
+    def text_update(self):
+        """
+        Updates the text on the screen.
+        """
+        self.textvariable.set(self.buffer)
+        if self.last_text:
+            self.delete(self.last_text)
+        kwargs = {
+            'anchor': tk.NW,
+            'fill': 'white',
+            'text': self.buffer,
+            'width': self['width'],
+            'font': self.font,
+        }
+        self.last_text = self.create_text(*self.line_pos, **kwargs)
+        return
+
+    def cursor_blink(self):
```

Comment:
This function should not be called by users, so I would suggest using the `_` prefix for it. I'd also use the prefix for the private attributes of this function, such as `cursor`, `blink` and others. This way, they are omitted from code-completion and the user is not tempted to use them or change them in ways they are not intended to be changed.